### PR TITLE
fixed the bug that hides new filter button

### PR DESCRIPTION
**Pull request by emre2038**
## Description
bug : 
add new filter button was not visible when 
* there are no default filters
* there are no non-aggregatable columns
* all the available filters are set as indicator-specific option

added a condition to make sure the add new filter is still visible 

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-675

## How is it tested automatically?
updated `__tests__/gui/sharing.feature`

## How should a reviewer test it locally
watch the video in the Jira card

### Updated
* `src/js/elements/subindicator_filter/filter_controller.js` to make sure the button is visible

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: src/js/elements/subindicator_filter/filter_controller.js [hunk 1]
```diff
@@ -175,8 +175,9 @@ export class FilterController extends Component {
     shouldFiltersBeVisible() {
         const nonAggregatableGroups = this.model.dataFilterModel.nonAggregatableGroups;
         const defaultGroups = this.model.dataFilterModel.defaultFilterGroups;
+        const previouslySelectedFilterGroups = this.model.dataFilterModel.previouslySelectedFilterGroups;
 
-        if (nonAggregatableGroups.length <= 0 && defaultGroups.length <= 0 && this.model.dataFilterModel.availableFilters.length <= 0) {
+        if (nonAggregatableGroups.length <= 0 && defaultGroups.length <= 0 && previouslySelectedFilterGroups.length <= 0 && this.model.dataFilterModel.availableFilters.length <= 0) {
```
Comment (michaelglenister) [inline src/js/elements/subindicator_filter/filter_controller.js:180]: Maybe assigning length to the constants could make the if statement a little neater. Not a big issue though

```
const previouslySelectedFilterGroups = this.model.dataFilterModel.previouslySelectedFilterGroups.length;
if (previouslySelectedFilterGroups <= 0) {...}
```

**Review by michaelglenister — APPROVED** [review]
Approved but there is a small suggestion that is sort of optional
